components/pkg-export-docker/defaults/Dockerfile.hbs

```diff
@@ -1,7 +1,10 @@
 FROM scratch
 ENV PATH {{path}}
+{{install_hook_env}}
+ENV OFFLINE_INSTALL ON
 ADD {{rootfs}} /
 
 EXPOSE 9631 {{exposes}}
+RUN {{hab_path}} pkg install {{installed_primary_svc_ident}}
```

Comment:
no. we load the packages with `InstallHookMode::ignore` while constructing the rootfs because we don't want the install hook to run on the host. If the INSTALL_HOOK feature is toggled (above),  then we toggle it here in the build so that this `pkg install` runs the install hook in the container. Yes, if it is not toggled then this is a no-op.

Reply:
In some of my other comments, I floated the possibility of pushing more of this toggling logic into the template itself. If we did that, both the setting of the environment variable and this `RUN` command could be moved into the same `if` block, making it very clear what the relationship between them is. It's not as clear now, given that they're physically distant in the Dockerfile.

Reply:
Another thing... if we set the environment variable in the Dockerfile, it will be enabled whenever the container is used, whereas if we supply it only in the `RUN` command, it would only be enabled for that one command.

That may be a better way to localize the effects of the flag, and might prevent unexpected behavior at runtime.

Reply:
Agree to both points.